src/Common/tests/gtest_lru_resource_cache.cpp

```diff
@@ -0,0 +1,276 @@
+#include <iomanip>
+#include <iostream>
+#include <gtest/gtest.h>
+#include <Common/LRUResourceCache.h>
+
+TEST(LRUResourceCache, acquire)
+{
+    using MyCache = DB::LRUResourceCache<int, int>;
+    auto mcache = MyCache(10, 10);
+    int x = 10;
+    auto load_int = [&] { return std::make_shared<int>(x); };
+    auto val = mcache.acquire(1, load_int);
+    x = 11;
+    val = mcache.acquire(2, load_int);
+    ASSERT_TRUE(val != nullptr);
+    ASSERT_TRUE(*val == 11);
+
+    val = mcache.acquire(1);
+    ASSERT_TRUE(val != nullptr);
+    ASSERT_TRUE(*val == 10);
+}
+
+TEST(LRUResourceCache, remove)
+{
+    using MyCache = DB::LRUResourceCache<int, int>;
+    auto mcache = MyCache(10, 10);
+    int x = 10;
+    auto load_int = [&] { return std::make_shared<int>(x); };
+    auto val = mcache.acquire(1, load_int);
+    x = 11;
+    val = mcache.acquire(2, load_int);
+
+    auto succ = mcache.tryRemove(3);
+    ASSERT_TRUE(succ);
+
+    succ = mcache.tryRemove(1);
+    ASSERT_TRUE(!succ);
+    val = mcache.acquire(1);
```

Comment:
Ok but I also stared to think that it makes sense to have a method (not sure how to name it, may be just `get`). It will return a holder, which will call `acquire` in constructor and `release` in desctructor. And may be make public only this method, `acquire` and `release` may be private. What do you think?

Reply:
good idea, I take a try

Reply:
updated